### PR TITLE
Upload test results based on actual jenkins build number

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
Looks like https://github.com/opensearch-project/opensearch-build/pull/2488 only fixed the message but not the actual path. This PR fixes the same for all tests.

The test_run_id was hardcoded to 1 for [integ-test ](https://github.com/opensearch-project/opensearch-build/blob/main/vars/runIntegTestScript.groovy#L24)and [bwc-test](https://github.com/opensearch-project/opensearch-build/blob/main/vars/runBwcTestScript.groovy#L15) which now is changed to build_number of the job.
This will allow us to run any number of tests for same build_id without overwriting the results.
In case of performance tests since it is not integrated with test_recorder yet, the build_number is added as a part of the upload path.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
The `--test-run-id` can have the `BUILD_NUMBER` for the bwc and integ tests. So now the prefix path would be `integ-test/1.3.5/5960/linux/arm64/tar/test-results/<BUILD_NUMBER>`

**Comment by gaiksaya** [comment]
Hi @prudhvigodithi please review. I made the changes. Thanks!

File: jenkins/opensearch-dashboards/integ-test.jenkinsfile [hunk 1]
```diff
@@ -98,7 +98,7 @@ pipeline {
                         uploadTestResults(
                             buildManifestFileName: BUILD_MANIFEST,
                             jobName: JOB_NAME
-                        )
+                            )
```
Comment (peterzhuamazon) [inline jenkins/opensearch-dashboards/integ-test.jenkinsfile:101]: Extra space?

File: jenkins/opensearch-dashboards/bwc-test.jenkinsfile [hunk 2]
```diff
@@ -101,9 +101,8 @@ pipeline {
                     script {
                         uploadTestResults(
                             buildManifestFileName: BUILD_MANIFEST,
-                            jobName: JOB_NAME,
-                            buildNumber: BUILD_ID
-                        )
+                            jobName: JOB_NAME
+                            )
```
Comment (peterzhuamazon) [inline jenkins/opensearch-dashboards/bwc-test.jenkinsfile:105]: Extra Space?

**Review by peterzhuamazon — APPROVED** [review]
Just some spacing issues I am good with this change.
Thanks.

File: jenkins/opensearch-dashboards/bwc-test.jenkinsfile (hunk 2 above)
Comment (gaiksaya) [reply jenkins/opensearch-dashboards/bwc-test.jenkinsfile]: Fixed it. Thanks!

**Pull request merged**